sdk/java/dagger-java-annotation-processor/src/test/resources/io/dagger/gen/entrypoint/entrypoint.java

```diff
@@ -195,17 +208,17 @@ private JSON invoke(JSON parentJson, String parentName, String fnName,
         DaggerJava obj = (DaggerJava) JsonConverter.fromJSON(parentJson, clazz);
         String[] stringArray = null;
         if (inputArgs.get("stringArray") != null) {
-          stringArray = (String[]) JsonConverter.fromJSON(inputArgs.get("stringArray"), String[].class);
+          stringArray = JsonConverter.fromJSON(inputArgs.get("stringArray"), String[].class);
         }
         Objects.requireNonNull(stringArray, "stringArray must not be null");
-        List ints = null;
+        List<Integer> ints = null;
         if (inputArgs.get("ints") != null) {
-          ints = (List) JsonConverter.fromJSON(inputArgs.get("ints"), List.class);
+          ints = Arrays.asList(JsonConverter.fromJSON(inputArgs.get("ints"), Integer[].class));
         }
```

Comment:
A lot of great things here, and a good example of the improvements regarding types:

`List ints` replaced by `List<Integer>`: the inner type is now better handled. So we are dealing with typed list and not just generic `List`.

The input is a JSON array of integers, but the end variable is a `List<Integer>`. So we would like to deserialize the input into the list. But from the JSON perspective it doesn't matter, this is an array.
And on the Java side, `Integer[].class` is what is the closest to the JSON type. `List.class` is generic and it's hard to correctly deserialize the JSON array into this type.

So the JSON array is deserialized into an array of the expected type (`Integer` here) then converted into a `List` so that the `List<Integer>` type of the variable is respected.